pkg/sql/opt/xform/testdata/rules/join

```diff
@@ -12378,3 +12378,37 @@ project
       │    └── columns: col0:8
       └── filters
            └── col3:4 = col0:8 [outer=(4,8), constraints=(/4: (/NULL - ]; /8: (/NULL - ]), fd=(4)==(8), (8)==(4)]
+
+# Regression for #81649
+exec-ddl
+CREATE TABLE t81649 (
+  col0 STRING NULL,
+  col1 DECIMAL NOT NULL,
+  col2 STRING NULL AS (lower(col0)) STORED,
+  col3 DECIMAL NOT NULL AS (col1 + 0:::DECIMAL) VIRTUAL,
```

Comment:
If you change the `VIRTUAL` to `STORED`, the problem reproduces on 21.2, so this is probably something we want to backport.